sia-gpu-miner.c

```diff
@@ -302,10 +313,33 @@ void printPlatformsAndDevices() {
 			ret = clGetDeviceInfo(deviceids[j], CL_DEVICE_NAME, 80, str, NULL);
 			if (ret != CL_SUCCESS) {
 				printf("\tError while getting device info.\n");
-				free(deviceids);
-				continue;
-			}
-			printf("\tDevice %d: %s\n", j, str);
+            } else {
+                printf("\tDevice %d: %s\n", j, str);
+            }
```

Comment:
if we can't read the device name, does it make sense to continue printing the endianness, alloc size, etc.? I figure if one fails, the rest will probably fail too. I'm not very familiar of OpenCL, though; maybe not all devices have names.